lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/CameraActivity.java

```diff
@@ -17,278 +17,307 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
 
-public abstract class CameraActivity extends AppCompatActivity
-    implements OnImageAvailableListener,
-        Camera.PreviewCallback,
-        CompoundButton.OnCheckedChangeListener,
-        View.OnClickListener {
+public class CameraActivity extends AppCompatActivity implements View.OnClickListener {
   private static final Logger LOGGER = new Logger();
 
   private static final int PERMISSIONS_REQUEST = 1;
 
   private static final String PERMISSION_CAMERA = Manifest.permission.CAMERA;
   protected int previewWidth = 0;
   protected int previewHeight = 0;
+
   private boolean debug = false;
+
   private Handler handler;
   private HandlerThread handlerThread;
-  private boolean useCamera2API;
+  private boolean firstTimeStartModel = true;
   private boolean isProcessingFrame = false;
-  private byte[][] yuvBytes = new byte[3][];
-  private int[] rgbBytes = null;
-  private int yRowStride;
-  private Runnable postInferenceCallback;
-  private Runnable imageConverter;
 
   private LinearLayout bottomSheetLayout;
-  private LinearLayout gestureLayout;
   private BottomSheetBehavior<LinearLayout> sheetBehavior;
 
-  protected TextView frameValueTextView, cropValueTextView, inferenceTimeTextView;
+  private enum DetectorMode {
+    TF_OD_API;
+  }
+
+  private static final int TF_OD_API_INPUT_SIZE = 300;
+  private static final boolean TF_OD_API_IS_QUANTIZED = true;
+  private static final String TF_OD_API_MODEL_FILE = "detect.tflite";
+  private static final String TF_OD_API_LABELS_FILE = "labelmap.txt";
+  private static final DetectorMode MODE = DetectorMode.TF_OD_API;
+
+  // Minimum detection confidence to track a detection.
+  private static final float MINIMUM_CONFIDENCE_TF_OD_API = 0.5f;
+  private static final Size DESIRED_ANALYSIS_SIZE = new Size(640, 480);
+  private static final float TEXT_SIZE_DIP = 10;
+  private long lastProcessingTimeMs;
+
   protected ImageView bottomSheetArrowImageView;
-  private ImageView plusImageView, minusImageView;
-  private SwitchCompat apiSwitchCompat;
-  private TextView threadsTextView;
+
+  private Integer sensorOrientation;
+
+  private Detector detector;
+
+  //Data Binding
+  private TfeOdActivityCameraBinding binding;
+
+  private long timestamp = 0;
+  private MultiBoxTracker tracker;
 
   @Override
   protected void onCreate(final Bundle savedInstanceState) {
     LOGGER.d("onCreate " + this);
     super.onCreate(null);
     getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
 
-    setContentView(R.layout.tfe_od_activity_camera);
-    Toolbar toolbar = findViewById(R.id.toolbar);
-    setSupportActionBar(toolbar);
-    getSupportActionBar().setDisplayShowTitleEnabled(false);
+    binding = DataBindingUtil.setContentView(this, R.layout.tfe_od_activity_camera);
 
     if (hasPermission()) {
-      setFragment();
+      //Start CameraX
```

Comment:
Comment should starts with a space, like `// Start CameraX`

Reply:
Noted!